### PR TITLE
docs: add submitting_block_process.md

### PR DESCRIPTION
**Pull request by keroro520**
Preview: https://github.com/keroro520/godwoken/blob/docs-submitting-block-process/docs/submitting_block_process.md

### CODE REVIEW

**Review by jjyr — DISMISSED** [review]

File: docs/submitting_block_process.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+- `l2block.raw.prev_account`
+  Assert that `l2block.raw.prev_account` must be equal to [`prev_global_state.account`](https://github.com/nervosnetwork/godwoken/blob/6148a733fcf7b41ce25aee3a44e8f2ae6158390d/crates/types/schemas/godwoken.mol#L28).
+
+- `l2block.raw.post_account`
+  `l2block.raw.post_account` represents the global accounts tree, also called the World State. As defined in the blockchain protocol, the previous World State will transit to the new World State by applying deposits/withdrawals/L2Transactions included in the present _l2block._
+
+  _State Validator_ is able to apply the deposits and withdrawals(read more from chapter "Apply Deposits and Withdrawals"). However, as an Optimistic Rollup framework, Godwoken does execute l2transactions on L1. This means the L1 scripts are not able to actually verify the _World State_ by directly applying l2transactions. Optimistic Rollup's challenge mechanism takes the responsibility to enforce the correctness of `L2Block.raw.post_account`.
```
Comment (blckngm) [inline docs/submitting_block_process.md]: `Godwoken does execute` -> `Godwoken does not execute`?

File: docs/submitting_block_process.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+- `l2block.raw.state_checkpoint_list`
+  `l2block.raw.state_checkpoint_list` contains a list of checkpoints that represents the _World State_ after applying every transactions. Therefore, its length must be equal to `l2block.transactions`.
```
Comment (blckngm) [inline docs/submitting_block_process.md:94]: Should we mention v2 changes about `state_checkpoint_list` here?

File: docs/submitting_block_process.md (hunk 1 above)
Comment (keroro520) [reply docs/submitting_block_process.md]: Updated. Thanks!

File: docs/submitting_block_process.md (hunk 2 above)
Comment (keroro520) [reply docs/submitting_block_process.md:94]: v2 does not change the structure or content of state_checkpoint_list, but only removes verifications for state_checkpoint_list. I prefer not to mention it here.

File: docs/submitting_block_process.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+  _State Validator_ is able to apply the deposits and withdrawals(read more from chapter "Apply Deposits and Withdrawals"). However, as an Optimistic Rollup framework, Godwoken does not execute l2transactions on L1. This means the L1 scripts are not able to actually verify the _World State_ by directly applying l2transactions. Optimistic Rollup's challenge mechanism takes the responsibility to enforce the correctness of `L2Block.raw.post_account`.
+
+- `l2block.raw.state_checkpoint_list`
+  `l2block.raw.state_checkpoint_list` contains a list of checkpoints that represents the _World State_ after applying every transactions. Therefore, its length must be equal to `l2block.transactions`.
+
+   Besides, Optimistic Rollup's challenge mechanism takes the responsibility to enforce the correctness of `l2block.raw.state_checkpoint_list`.
+
+- `l2block.raw.submit_withdrawals`
+  ❓
```
Comment (zeroqn) [inline docs/submitting_block_process.md:99]: How about this?
A commitment to withdrawals included in the block. It contains withdrawals count and withdrawals merkle root.

File: docs/submitting_block_process.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+- `l2block.raw.submit_transactions`
+  ❓
```
Comment (zeroqn) [inline docs/submitting_block_process.md:102]: A commitment to transactions included in the block. It contains transactions count and transactions merkle root.

**Comment by jjyr** [comment]
@sopium @zeroqn is this PR ok to merge?

I will take @zeroqn 's advice to update the doc in another PR.

**Review by blckngm — APPROVED** [review]

**Review by jjyr — APPROVED** [review]

**Pull request merged**